src/Chainweb/Mempool/InMem.hs

```diff
@@ -611,3 +624,36 @@ getMempoolStats m = do
         <$!> (HashMap.size <$!> readIORef (_inmemPending d))
         <*> (length . _rlRecent <$!> readIORef (_inmemRecentLog d))
         <*> (HashMap.size <$!> readIORef (_inmemBadMap d))
+
+------------------------------------------------------------------------------
+-- | Prune the mempool's pending map and badmap.
+pruneInMem
+    :: forall t . NFData t
+    => MVar (InMemoryMempoolData t)
+    -> IO ()
+pruneInMem lock = do
+    now <- getCurrentTimeIntegral
+    withMVar lock $ \mdata -> pruneInternal mdata now
+
+
+------------------------------------------------------------------------------
+pruneInternal
+    :: forall t . NFData t
+    => InMemoryMempoolData t
+    -> Time Micros
+    -> IO ()
+pruneInternal mdata now = do
+    let pref = _inmemPending mdata
+    !pending <- readIORef pref
+    !pending' <- evaluate $ force $ HashMap.filter flt pending
```

Comment:
How expensive is this and how often it is down? Would it make sense to use a better data structure for this? It would probably be overkill if this isn't called too often.